compiler/rustc_infer/src/infer/type_variable.rs

```diff
@@ -164,6 +231,18 @@ impl<'tcx> TypeVariableTable<'_, 'tcx> {
         self.eq_relations().find(vid).vid
     }
 
+    /// Returns the "root" variable of `vid` in the `sub_relations`
+    /// equivalence table. All type variables that have been are
+    /// related via equality or subtyping will yield the same root
+    /// variable (per the union-find algorithm), so `sub_root_var(a)
+    /// == sub_root_var(b)` implies that:
+    /// ```text
+    /// exists X. (a <: X || X <: a) && (b <: X || X <: b)
+    /// ```
+    pub(crate) fn sub_root_var(&mut self, vid: ty::TyVid) -> ty::TyVid {
```

Comment:
similarly `sub_root_var` is confusing. it makes it sound like its the root of a directed graph of subtyping relations which then makes me start thinking about what cases that can get invalidated :>

but in reality nothing here is directed so we can just arbitrarily pick some var as the root and its fine forever. no idea what a better name is

Reply:
`sub_relations_representative` maybe. Renamed it to `sub_relations_root_var` for now